src/shared/treeview/utils.ts

```diff
@@ -70,3 +72,34 @@ export function createErrorItem(error: Error, message?: string): AWSTreeNodeBase
         contextValue: 'awsErrorNode',
     })
 }
+
+export function createPlaceholderItem(message: string): TreeNode {
+    return {
+        id: 'placeholder',
+        resource: message,
+        treeItem: new vscode.TreeItem(message, vscode.TreeItemCollapsibleState.None),
+    }
+}
+
+export function unboxTreeNode<T>(node: TreeNode, predicate: (resource: unknown) => resource is T): T {
+    if (!predicate(node.resource)) {
+        throw new TypeError(`Unexpected tree node resource for node: ${node.id}`)
+    }
+
+    return node.resource
+}
+
+export class TreeShim extends AWSTreeNodeBase {
```

Comment:
Very simple class so `TreeNode` can be used with legacy code. Entry-points that consume nodes would need to take this into account though.